src/clients/c++/library/http_client.cc

```diff
@@ -1080,8 +1080,12 @@ InferenceServerHttpClient::Infer(
 
   // During this call SEND_END (except in above case), RECV_START, and
   // RECV_END will be set.
-  if (curl_easy_perform(easy_handle_) != CURLE_OK) {
+  auto curl_status = curl_easy_perform(easy_handle_);
+  if (curl_status != CURLE_OK) {
     sync_request->http_code_ = 400;
```

Comment:
So in a timeout the client just sees a 400 code. Seems like they need to be able to differentiate a timeout from other 400 failures. There isn't a good HTTP code for this so we may just have to use 499 or something.